migrations/sqls/20190324132907-dbdev-up.sql

```diff
@@ -0,0 +1,182 @@
+    firstName varchar(64) NOT NULL,
+    lastName varchar(64) NOT NULL,
+    accessLevel nvarchar(32) NOT NULL, 
+    userName nvarchar(64) NOT NULL,
+    password nvarchar(64) NOT NULL
+);
+
+/*
+Create lead table: Here things get a bit interesting, leads and people are siloed but there is potential for leads to become people, and it would be good to track that..
+*/
+CREATE TABLE IF NOT EXISTS lead (
+	leadId int AUTO_INCREMENT PRIMARY KEY,
+    firstName varchar(64) NOT NULL,
+    lastName varchar(64) NOT NULL,
+    leadLocation varchar(128),
+    eventName varchar(128),
+    prmaryContact varchar(32) NOT NULL,
+    zipCode varchar(5) NOT NULL,
+    gender varchar(32),
+    dateAdded DATETIME DEFAULT CURRENT_TIMESTAMP,
+    addedBy int NOT NULL,
+    FOREIGN KEY (addedBy) REFERENCES person(personId)
+);
+/*
+Create Programs: CU you can define its programs and programs managers in this table.
+*/
+CREATE TABLE IF NOT EXISTS programs (
+	programId int AUTO_INCREMENT PRIMARY KEY,
+    programName varchar(32) NOT NULL,
+    description varchar(128),
+    programManager int,
+    createdDate DATETIME DEFAULT CURRENT_TIMESTAMP,
+    programStatus varchar(32),
+    FOREIGN KEY (programManager) REFERENCES person(personId)
+);
+/*
+Create Services Table: Services are in turn adminstered by a program. Pogram 1:M Services
+*/
+CREATE TABLE IF NOT EXISTS servcies (
+	serviceId int AUTO_INCREMENT PRIMARY KEY,
+    programId INT NOT NULL,
+    serviceName varchar(64),
+    serviceDesc varchar(128),
+    FOREIGN KEY (programId) REFERENCES programs(programId)
+);
+/*
+Create LeadServicesNeeded: When a record is added to leads table it may require more than once serives. This table tracks all services required by each lead.. lead 1:M leadServicesNeeded
+*/
+CREATE TABLE IF NOT EXISTS leadServicesNeeded (
+	leadServicedNeededId int AUTO_INCREMENT PRIMARY KEY,
+	leadId int NOT NULL,
+    serviceId int NOT NULL,
+    comments varchar(128),
+    dateAdded DATETIME DEFAULT CURRENT_TIMESTAMP,
+    FOREIGN KEY (sericeId) REFERENCES services(serviceId),
+    FOREIGN KEY (leadId) REFERENCES lead(leadId)
+);
+
+/*
+FROM HERE ON, THE DATABASE IS GETTING INTO MORE DETAIL AND THIS MAYBE PART OF ADDED MODELS.. FOR DISCUSSION
+Create Person Lead to Client (When a new a lead becomes a client lead first name, 
+last name and gender are turned over to person table, leadToPerson gets an insert to maintain that record)
+leadToClient 1:1 Person
+*/
+CREATE TABLE IF NOT EXISTS leadToClient (
+	leadToClientId int AUTO_INCREMENT PRIMARY KEY,
+    personId int NOT NULL,
+    leadId int NOT NULL,
+    comments varchar(5000),
+    dateLeadToClient date,
+    FOREIGN KEY (personId) REFERENCES person(personId),
+    FOREIGN KEY (leadId) REFERENCES lead(leadId)
+);
+/*
+Create IntakeData Table: Intake data is collected on client inital and subsequent visits to CU I expect this to change quite a bit based on feed back form CU
+*/
+CREATE TABLE IF NOT EXISTS intakeData (
```

Comment:
This is an interesting table -- talking with Vicky and Luis a couple weeks ago it sounds like the long term hope is to be able to track the time spent with clients. But that we wouldn't start with that.

This table seems to be a combination of tracking visits and capturing one-time information. I think it might make most sense just to capture the one time information for now, with the plan to add more to the database schema once we want to track visits.